### PR TITLE
[Merged by Bors] - feat: Fin.exists_fin_succ'

### PR DESCRIPTION
**Pull request by enricozb**
Adds `Fin.exists_fin_succ'`, which is analogous to [`Fin.forall_fin_succ'`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fin/Basic.html#Fin.forall_fin_succ').

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/Basic.lean [hunk 1]
```diff
@@ -996,6 +996,11 @@ theorem forall_fin_succ' {P : Fin (n + 1) → Prop} :
 theorem eq_castSucc_or_eq_last {n : Nat} (i : Fin (n + 1)) :
     (∃ j : Fin n, i = j.castSucc) ∨ i = last n := i.lastCases (Or.inr rfl) (Or.inl ⟨·, rfl⟩)
 
+theorem exists_fin_succ' {P : Fin (n + 1) → Prop} :
+    (∃ i, P i) ↔ (∃ i : Fin n, P i.castSucc) ∨ P (.last _) :=
+  ⟨fun ⟨i, h⟩ => Fin.lastCases Or.inr (fun i hi => Or.inl ⟨i, hi⟩) i h, fun h =>
+    h.elim (fun ⟨i, hi⟩ => ⟨i.castSucc, hi⟩) (fun h => ⟨.last _, h⟩)⟩
```
Comment (Vierkantor) [inline Mathlib/Data/Fin/Basic.lean:1002]: Let's wrap the lines slightly differently so the directions of the iff each have their own line:
```suggestion
  ⟨fun ⟨i, h⟩ => Fin.lastCases Or.inr (fun i hi => Or.inl ⟨i, hi⟩) i h,
   fun h => h.elim (fun ⟨i, hi⟩ => ⟨i.castSucc, hi⟩) (fun h => ⟨.last _, h⟩)⟩
```

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Data/Fin/Basic.lean (hunk 1 above)
Comment (enricozb) [reply Mathlib/Data/Fin/Basic.lean:1002]: Good suggestion, thanks :)

**Pull request closed without merging**